src/python/BuildingControlsSimulator/DataClients/DataClient.py

```diff
@@ -48,267 +51,137 @@ def make_data_directories(self):
         os.makedirs(self.ep_tmy3_cache_dir, exist_ok=True)
         os.makedirs(self.simulation_epw_dir, exist_ok=True)
 
-    def get_data(self, tstat_sim_config):
-
+    def get_data(self):
         # check for invalid start/end combination
-        invalid = tstat_sim_config[
-            tstat_sim_config["end_utc"] <= tstat_sim_config["start_utc"]
-        ]
-
-        if not invalid.empty:
+        if self.sim_config["end_utc"] <= self.sim_config["start_utc"]:
             raise ValueError(
-                "tstat_sim_config contains invalid start_utc >= end_utc."
+                "sim_config contains invalid start_utc >= end_utc."
             )
-
-        _data = {
-            identifier: pd.DataFrame([], columns=[Internal.datetime_column])
-            for identifier in tstat_sim_config.index
-        }
-        for _s in self.sources:
-            # load from cache or download data from source
-            _data_dict = _s.get_data(tstat_sim_config)
-            for tstat, _df in _data_dict.items():
-                # joining on datetime column with the initial empty df having
-                # only the datetime_column causes new columns to be added
-                # and handles missing data in any data sets
-                if not _df.empty:
-                    _data[tstat] = _data[tstat].merge(
-                        _df, how="outer", on=Internal.datetime_column,
-                    )
-                else:
-                    logging.info(
-                        "EMPTY SOURCE: tstat={}, source={}".format(tstat, _s)
-                    )
-                    if _data[tstat].empty:
-                        _data[tstat] = Internal.get_empty_df()
+        # load from cache or download data from source
+        _data = self.source.get_data(self.sim_config)
+        _data = _data.sort_index()
+
+        if _data.empty:
+            logging.error(
+                "EMPTY DATA SOURCE: \nsim_config={} \nsource={}\n".format(
+                    self.sim_config, self.source
+                )
+            )
+            if _data.empty:
```

Comment:
Seems redundant with the if in the outer scope.

Reply:
yupe this should be removed.